Features/Contacts/Sources/Scenes/ContactAddressListScene.swift

```diff
@@ -0,0 +1,127 @@
+ //Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUI
+import Style
+import Components
+import Store
+import GRDBQuery
+import PrimitivesComponents
+import Primitives
+import Localization
+
+public struct ContactAddressListScene: View {
+    
+    private var model: ContactAddressListViewModel
+    
+    @Query<ContactAddressListRequest>
+    private var addresses: [ContactAddress]
+    
+    @State var addressToDelete: ContactAddress? = .none
+    @State private var presentingErrorMessage: String?
+    @Binding var isPresentingContactAddressInput: AddContactAddressInput?
+    
+    public init(
+        model: ContactAddressListViewModel,
+        isPresentingContactAddressInput: Binding<AddContactAddressInput?>
+    ) {
+        self.model = model
+        _isPresentingContactAddressInput = isPresentingContactAddressInput
+        
+        let request = Binding {
+            model.addressListRequest
+        } set: { new in
+            model.addressListRequest = new
+        }
+        
+        _addresses = Query(request)
+    }
+    
+    public var body: some View {
+        VStack {
+            if addresses.isEmpty {
+                StateEmptyView(
+                    title: "Address list is empty",
+                    description: "Tap + to add the first one."
+                )
+            } else {
+                listView
+            }
+        }
+        .navigationTitle(model.title)
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
+                Button {
+                    isPresentingContactAddressInput = model.input(from: nil)
+                } label: {
+                    Images.System.plus
+                }
+            }
+        }
+        .alert(
+            "",
+            isPresented: $presentingErrorMessage.mappedToBool(),
```

Comment:
```suggestion
            isPresented: $isPresentingErrorMessage.mappedToBool(),
```

consistency across the app